src/Shared/Targets/Wrappers/AspNetBufferingTargetWrapper.cs

```diff
@@ -193,12 +194,34 @@ protected override void CloseTarget()
             base.CloseTarget();
         }
 
+        private static bool LogEventProcessed;
+        private static readonly object LogEventProcessedLock = new object();
+
         /// <summary>
         /// Adds the specified log event to the buffer.
         /// </summary>
         /// <param name="logEvent">The log event.</param>
         protected override void WriteAsyncThreadSafe(AsyncLogEventInfo logEvent)
         {
+            if (!LogEventProcessed)
+            {
+                lock (LogEventProcessedLock)
```

Comment:
Though I like the idea of only printing the alert exactly once, then I think using a monitor-lock-object is a little too heavy. Could it just be (Removing `LogEventProcessedLock`):
```c#
if (!LogEventProcessed)
{
   LogEventProcessed = true;
   if (!MiddlewareInstalled)
   {
       ...
   }
}
```